apiserver/facades/client/controller/controller.go

```diff
@@ -544,40 +475,35 @@ func (c *ControllerAPI) HostedModelConfigs(ctx context.Context) (params.HostedMo
 		return result, errors.Trace(err)
 	}
 
-	modelUUIDs, err := c.state.AllModelUUIDs()
+	models, err := c.modelService.ListAllModels(ctx)
 	if err != nil {
-		return result, errors.Trace(err)
+		return result, fmt.Errorf("getting model IDs: %w", err)
+	}
+	controllerModel, err := c.modelService.ControllerModel(ctx)
+	if err != nil {
+		return result, fmt.Errorf("getting controller model: %w", err)
 	}
 
-	for _, modelUUID := range modelUUIDs {
-		if modelUUID == c.state.ControllerModelUUID() {
+	for _, model := range models {
+		if model.UUID == controllerModel.UUID {
 			continue
 		}
-		st, err := c.statePool.Get(modelUUID)
-		if err != nil {
-			// This model could have been removed.
-			if errors.Is(err, errors.NotFound) {
-				continue
-			}
-			return result, errors.Trace(err)
-		}
-		defer st.Release()
-		model, err := st.Model()
-		if err != nil {
-			return result, errors.Trace(err)
-		}
 
 		config := params.HostedModelConfig{
-			Name:     model.Name(),
-			OwnerTag: model.Owner().String(),
+			Name:     model.Name,
+			OwnerTag: names.NewUserTag(model.OwnerName).String(),
 		}
-		modelConf, err := model.Config()
+
+		modelConf, err := c.modelConfigServiceGetter(model.UUID).ModelConfig(ctx)
 		if err != nil {
-			config.Error = apiservererrors.ServerError(err)
-		} else {
-			config.Config = modelConf.AllAttrs()
+			// This model could have been removed.
+			if !errors.Is(err, modelerrors.NotFound) {
+				config.Error = apiservererrors.ServerError(err)
```

Comment:
You decided to change to continue whenever an error occurs.
The below `if config.Error == nil` check is unnecessary anymore.